sklearn/decomposition/_fastica.py

```diff
@@ -574,12 +574,13 @@ def g(x, fun_args):
                     )
                 d[degenerate_idx] = eps  # For numerical issues
                 d = np.sqrt(d, d)
-                d, u = d[sort_indices], u[sort_indices]
-                # Resize and reorder to match svd
-                u = u[::-1, : min(X.shape) : -1]
+                d, u = d[sort_indices], u[:, sort_indices]
```

Comment:
Based on [eigh](https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.eigh.html) docs, the eigenvectors has shape `(M, N)` where `N` is the number of eigenvalues. Thus, I think the sorting should be done over `axis=1`.